### ISSUE TITLE
Queries in quick succession being ignored with no error in windows

### ISSUE THREAD

**Issue by filipesbragio** [rqlite/rqlite#1262]
**What version are you running?**
7.14.3

**Are you using Docker or Kubernetes to run your system?**
no

**Are you running a single node or a cluster?**
Cluster with 3 VM's running Windows Server 2019 Standard. These VM's are configured to only have one processor.

**What did you do?**
Executed the following query in quick succession

"insert into [SomeTable] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value","AnotherField":2}

**What did you expect to happen?**
All inserts should have succeeded and be added to the database

**What happened instead?**
I've run this multiple times and intermittently some values don't get added to the database. In the case in the logs below four of the 67 inserts did not execute even though the response was 200.

I've also created a test to try to reproduce this on my machine but running the test locally with a cluster of 3 hundreds of times did not reproduce the issue.

I plan on fixing this by simply improving the queries to be inserted at once instead of individually, however I'm concerned this might happen to other unrelated queries that are run in quick succession.

Might be related to #1123 although we aren't using queued writes.

queries
```
18 May 2023 13:57:07,175 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value1","AnotherField":2}]]
18 May 2023 13:57:07,175 11   True
18 May 2023 13:57:07,175 11   {"results":[{"last_insert_id":123,"rows_affected":1}],"time":0.0069277}
18 May 2023 13:57:07,187 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value2","AnotherField":2}]]
18 May 2023 13:57:07,187 11   True
18 May 2023 13:57:07,187 11   {"results":[{"last_insert_id":124,"rows_affected":1}],"time":0.0096777}
18 May 2023 13:57:07,195 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value3","AnotherField":2}]]
18 May 2023 13:57:07,195 11   True
18 May 2023 13:57:07,195 11   {"results":[{"last_insert_id":125,"rows_affected":1}],"time":0.0067188}
18 May 2023 13:57:07,203 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value4","AnotherField":2}]]
18 May 2023 13:57:07,203 11   True
18 May 2023 13:57:07,203 11   {"results":[{"last_insert_id":126,"rows_affected":1}],"time":0.0078791}
18 May 2023 13:57:07,213 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value5","AnotherField":2}]]
18 May 2023 13:57:07,213 11   True
18 May 2023 13:57:07,213 11   {"results":[{"last_insert_id":127,"rows_affected":1}],"time":0.0097367}
18 May 2023 13:57:07,223 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value6","AnotherField":2}]]
18 May 2023 13:57:07,223 11   True
18 May 2023 13:57:07,223 11   {"results":[{"last_insert_id":128,"rows_affected":1}],"time":0.0092587}
18 May 2023 13:57:07,232 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value7","AnotherField":1}]]
18 May 2023 13:57:07,232 11   True
18 May 2023 13:57:07,232 11   {"results":[{"last_insert_id":129,"rows_affected":1}],"time":0.0079167}
18 May 2023 13:57:07,241 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value8","AnotherField":2}]]
18 May 2023 13:57:07,241 11   True
18 May 2023 13:57:07,241 11   {"results":[{"last_insert_id":130,"rows_affected":1}],"time":0.008314}
18 May 2023 13:57:07,250 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value9","AnotherField":2}]]
18 May 2023 13:57:07,250 11   True
18 May 2023 13:57:07,250 11   {"results":[{"last_insert_id":131,"rows_affected":1}],"time":0.0074707}
18 May 2023 13:57:07,258 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value10","AnotherField":2}]]
18 May 2023 13:57:07,258 11   True
18 May 2023 13:57:07,258 11   {"results":[{"last_insert_id":132,"rows_affected":1}],"time":0.0073423}
18 May 2023 13:57:07,266 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value11","AnotherField":2}]]
18 May 2023 13:57:07,266 11   True
18 May 2023 13:57:07,266 11   {"results":[{"last_insert_id":133,"rows_affected":1}],"time":0.0084081}
18 May 2023 13:57:07,275 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value12","AnotherField":2}]]
18 May 2023 13:57:07,275 11   True
18 May 2023 13:57:07,275 11   {"results":[{"last_insert_id":134,"rows_affected":1}],"time":0.0076352}
18 May 2023 13:57:07,284 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value13","AnotherField":2}]]
18 May 2023 13:57:07,284 11   True
18 May 2023 13:57:07,284 11   {"results":[{"last_insert_id":135,"rows_affected":1}],"time":0.0094137}
18 May 2023 13:57:07,293 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value14","AnotherField":2}]]
18 May 2023 13:57:07,293 11   True
18 May 2023 13:57:07,293 11   {"results":[{"last_insert_id":136,"rows_affected":1}],"time":0.0078132}
18 May 2023 13:57:07,301 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value15","AnotherField":2}]]
18 May 2023 13:57:07,301 11   True
18 May 2023 13:57:07,301 11   {"results":[{"last_insert_id":137,"rows_affected":1}],"time":0.0079536}
18 May 2023 13:57:07,309 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value16","AnotherField":2}]]
18 May 2023 13:57:07,309 11   True
18 May 2023 13:57:07,309 11   {"results":[{"last_insert_id":138,"rows_affected":1}],"time":0.0079663}
18 May 2023 13:57:07,322 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value17","AnotherField":2}]]
18 May 2023 13:57:07,322 11   True
18 May 2023 13:57:07,322 11   {"results":[{"last_insert_id":139,"rows_affected":1}],"time":0.0117764}
18 May 2023 13:57:07,339 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value18","AnotherField":2}]]
18 May 2023 13:57:07,339 11   True
18 May 2023 13:57:07,339 11   {"results":[{"last_insert_id":140,"rows_affected":1}],"time":0.0162349}
18 May 2023 13:57:07,350 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value19","AnotherField":2}]]
18 May 2023 13:57:07,350 11   True
18 May 2023 13:57:07,350 11   {"results":[{"last_insert_id":141,"rows_affected":1}],"time":0.0095841}
18 May 2023 13:57:07,360 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value20","AnotherField":2}]]
18 May 2023 13:57:07,360 11   True
18 May 2023 13:57:07,360 11   {"results":[{"last_insert_id":142,"rows_affected":1}],"time":0.0088969}
18 May 2023 13:57:07,370 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value21","AnotherField":2}]]
18 May 2023 13:57:07,370 11   True
18 May 2023 13:57:07,370 11   {"results":[{"last_insert_id":143,"rows_affected":1}],"time":0.0087607}
18 May 2023 13:57:07,378 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value22","AnotherField":2}]]
18 May 2023 13:57:07,378 11   True
18 May 2023 13:57:07,378 11   {"results":[{"last_insert_id":144,"rows_affected":1}],"time":0.0073205}
18 May 2023 13:57:07,387 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value23","AnotherField":2}]]
18 May 2023 13:57:07,387 11   True
18 May 2023 13:57:07,387 11   {"results":[{"last_insert_id":145,"rows_affected":1}],"time":0.0087463}
18 May 2023 13:57:07,397 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value24","AnotherField":2}]]
18 May 2023 13:57:07,397 11   True
18 May 2023 13:57:07,397 11   {"results":[{"last_insert_id":146,"rows_affected":1}],"time":0.0095005}
18 May 2023 13:57:07,405 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value25","AnotherField":2}]]
18 May 2023 13:57:07,405 11   True
18 May 2023 13:57:07,405 11   {"results":[{"last_insert_id":147,"rows_affected":1}],"time":0.0071492}
18 May 2023 13:57:07,416 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value26","AnotherField":2}]]
18 May 2023 13:57:07,416 11   True
18 May 2023 13:57:07,416 11   {"results":[{"last_insert_id":148,"rows_affected":1}],"time":0.0091899}
18 May 2023 13:57:07,428 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value27","AnotherField":2}]]
18 May 2023 13:57:07,428 11   True
18 May 2023 13:57:07,428 11   {"results":[{"last_insert_id":149,"rows_affected":1}],"time":0.0107125}
18 May 2023 13:57:07,439 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value28","AnotherField":2}]]
18 May 2023 13:57:07,439 11   True
18 May 2023 13:57:07,439 11   {"results":[{"last_insert_id":150,"rows_affected":1}],"time":0.0094293}
18 May 2023 13:57:07,451 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value29","AnotherField":2}]]
18 May 2023 13:57:07,451 11   True
18 May 2023 13:57:07,451 11   {"results":[{"last_insert_id":151,"rows_affected":1}],"time":0.0108022}
18 May 2023 13:57:07,467 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value30","AnotherField":2}]]
18 May 2023 13:57:07,467 11   True
18 May 2023 13:57:07,467 11   {"results":[{"last_insert_id":152,"rows_affected":1}],"time":0.0156927}
18 May 2023 13:57:07,482 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value31","AnotherField":2}]]
18 May 2023 13:57:07,482 11   True
18 May 2023 13:57:07,482 11   {"results":[{"last_insert_id":153,"rows_affected":1}],"time":0.0140328}
18 May 2023 13:57:07,504 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value32","AnotherField":2}]]
18 May 2023 13:57:07,504 11   True
18 May 2023 13:57:07,504 11   {"results":[{"last_insert_id":154,"rows_affected":1}],"time":0.0207344}
18 May 2023 13:57:07,517 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value33","AnotherField":2}]]
18 May 2023 13:57:07,517 11   True
18 May 2023 13:57:07,517 11   {"results":[{"last_insert_id":155,"rows_affected":1}],"time":0.0120756}
18 May 2023 13:57:07,530 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value34","AnotherField":2}]]
18 May 2023 13:57:07,530 11   True
18 May 2023 13:57:07,530 11   {"results":[{"last_insert_id":156,"rows_affected":1}],"time":0.0117763}
18 May 2023 13:57:07,558 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value35","AnotherField":2}]]
18 May 2023 13:57:07,558 11   True
18 May 2023 13:57:07,558 11   {"results":[{"last_insert_id":157,"rows_affected":1}],"time":0.0274636}
18 May 2023 13:57:07,572 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value36","AnotherField":2}]]
18 May 2023 13:57:07,572 11   True
18 May 2023 13:57:07,572 11   {"results":[{"last_insert_id":158,"rows_affected":1}],"time":0.0125707}
18 May 2023 13:57:07,588 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value37","AnotherField":2}]]
18 May 2023 13:57:07,588 11   True
18 May 2023 13:57:07,588 11   {"results":[{"last_insert_id":159,"rows_affected":1}],"time":0.0146158}
18 May 2023 13:57:07,601 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value38","AnotherField":2}]]
18 May 2023 13:57:07,602 11   True
18 May 2023 13:57:07,602 11   {"results":[{"last_insert_id":160,"rows_affected":1}],"time":0.0132519}
18 May 2023 13:57:07,610 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value39","AnotherField":2}]]
18 May 2023 13:57:07,610 11   True
18 May 2023 13:57:07,610 11   {"results":[{"last_insert_id":161,"rows_affected":1}],"time":0.007198}
18 May 2023 13:57:07,619 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value40","AnotherField":2}]]
18 May 2023 13:57:07,619 11   True
18 May 2023 13:57:07,619 11   {"results":[{"last_insert_id":162,"rows_affected":1}],"time":0.0083746}
18 May 2023 13:57:07,628 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value41","AnotherField":2}]]
18 May 2023 13:57:07,628 11   True
18 May 2023 13:57:07,628 11   {"results":[{"last_insert_id":163,"rows_affected":1}],"time":0.007962}
18 May 2023 13:57:07,641 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value42","AnotherField":2}]]
18 May 2023 13:57:07,641 11   True
18 May 2023 13:57:07,641 11   {"results":[{"last_insert_id":164,"rows_affected":1}],"time":0.0116273}
18 May 2023 13:57:07,659 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value43","AnotherField":2}]]
18 May 2023 13:57:07,659 11   True
18 May 2023 13:57:07,659 11   {"results":[{"last_insert_id":165,"rows_affected":1}],"time":0.0168208}
18 May 2023 13:57:07,671 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value44","AnotherField":2}]]
18 May 2023 13:57:07,671 11   True
18 May 2023 13:57:07,671 11   {"results":[{"last_insert_id":166,"rows_affected":1,"time":0.0002417}],"time":0.0111244}
18 May 2023 13:57:07,689 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value45","AnotherField":2}]]
18 May 2023 13:57:07,689 11   True
18 May 2023 13:57:07,689 11   {"results":[{"last_insert_id":167,"rows_affected":1}],"time":0.0174886}
18 May 2023 13:57:07,704 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value46","AnotherField":2}]]
18 May 2023 13:57:07,704 11   True
18 May 2023 13:57:07,704 11   {"results":[{"last_insert_id":168,"rows_affected":1}],"time":0.0132057}
18 May 2023 13:57:07,727 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value47","AnotherField":2}]]
18 May 2023 13:57:07,727 11   True
18 May 2023 13:57:07,727 11   {"results":[{"last_insert_id":123,"rows_affected":1}],"time":0.0214365}
18 May 2023 13:57:07,747 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value48","AnotherField":2}]]
18 May 2023 13:57:07,747 11   True
18 May 2023 13:57:07,747 11   {"results":[{"last_insert_id":124,"rows_affected":1}],"time":0.018743}
18 May 2023 13:57:07,762 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value49","AnotherField":2}]]
18 May 2023 13:57:07,762 11   True
18 May 2023 13:57:07,762 11   {"results":[{"last_insert_id":125,"rows_affected":1}],"time":0.0141968}
18 May 2023 13:57:07,778 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value50","AnotherField":2}]]
18 May 2023 13:57:07,778 11   True
18 May 2023 13:57:07,778 11   {"results":[{"last_insert_id":126,"rows_affected":1}],"time":0.0155787}
18 May 2023 13:57:07,792 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value51","AnotherField":2}]]
18 May 2023 13:57:07,792 11   True
18 May 2023 13:57:07,792 11   {"results":[{"last_insert_id":127,"rows_affected":1}],"time":0.012229}
18 May 2023 13:57:07,806 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value52","AnotherField":2}]]
18 May 2023 13:57:07,806 11   True
18 May 2023 13:57:07,806 11   {"results":[{"last_insert_id":128,"rows_affected":1}],"time":0.0132766}
18 May 2023 13:57:07,833 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value53","AnotherField":2}]]
18 May 2023 13:57:07,833 11   True
18 May 2023 13:57:07,833 11   {"results":[{"last_insert_id":130,"rows_affected":1}],"time":0.0254148}
18 May 2023 13:57:07,851 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value54","AnotherField":2}]]
18 May 2023 13:57:07,851 11   True
18 May 2023 13:57:07,851 11   {"results":[{"last_insert_id":132,"rows_affected":1}],"time":0.0162226}
18 May 2023 13:57:07,868 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value55","AnotherField":2}]]
18 May 2023 13:57:07,868 11   True
18 May 2023 13:57:07,868 11   {"results":[{"last_insert_id":134,"rows_affected":1}],"time":0.0163527}
18 May 2023 13:57:07,885 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value56","AnotherField":2}]]
18 May 2023 13:57:07,885 11   True
18 May 2023 13:57:07,885 11   {"results":[{"last_insert_id":136,"rows_affected":1}],"time":0.0160599}
18 May 2023 13:57:07,903 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value57","AnotherField":2}]]
18 May 2023 13:57:07,903 11   True
18 May 2023 13:57:07,903 11   {"results":[{"last_insert_id":138,"rows_affected":1}],"time":0.0169019}
18 May 2023 13:57:07,921 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value58","AnotherField":2}]]
18 May 2023 13:57:07,921 11   True
18 May 2023 13:57:07,921 11   {"results":[{"last_insert_id":140,"rows_affected":1}],"time":0.0175673}
18 May 2023 13:57:07,942 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value59","AnotherField":2}]]
18 May 2023 13:57:07,942 11   True
18 May 2023 13:57:07,942 11   {"results":[{"last_insert_id":142,"rows_affected":1}],"time":0.019141}
18 May 2023 13:57:07,960 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value60","AnotherField":2}]]
18 May 2023 13:57:07,960 11   True
18 May 2023 13:57:07,960 11   {"results":[{"last_insert_id":144,"rows_affected":1}],"time":0.0170777}
18 May 2023 13:57:07,976 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value61","AnotherField":2}]]
18 May 2023 13:57:07,976 11   True
18 May 2023 13:57:07,976 11   {"results":[{"last_insert_id":146,"rows_affected":1}],"time":0.0155176}
18 May 2023 13:57:07,994 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value62","AnotherField":2}]]
18 May 2023 13:57:07,994 11   True
18 May 2023 13:57:07,994 11   {"results":[{"last_insert_id":148,"rows_affected":1}],"time":0.0161689}
18 May 2023 13:57:08,010 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value63","AnotherField":2}]]
18 May 2023 13:57:08,010 11   True
18 May 2023 13:57:08,010 11   {"results":[{"last_insert_id":150,"rows_affected":1}],"time":0.0155986}
18 May 2023 13:57:08,027 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value64","AnotherField":2}]]
18 May 2023 13:57:08,027 11   True
18 May 2023 13:57:08,027 11   {"results":[{"last_insert_id":152,"rows_affected":1}],"time":0.0154398}
18 May 2023 13:57:08,043 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value65","AnotherField":2}]]
18 May 2023 13:57:08,043 11   True
18 May 2023 13:57:08,043 11   {"results":[{"last_insert_id":154,"rows_affected":1}],"time":0.0156543}
18 May 2023 13:57:08,061 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value66","AnotherField":2}]]
18 May 2023 13:57:08,061 11   True
18 May 2023 13:57:08,061 11   {"results":[{"last_insert_id":156,"rows_affected":1}],"time":0.0161781}
18 May 2023 13:57:08,082 11   [["insert into [RoleFeaturePermission] (SomeIdField, OtherIdField, AnotherField) values (:SomeIdField, :OtherIdField, :AnotherField)", {"SomeIdField":"e47ef6af-fd0a-4208-bdaf-1b8e5d09678b","OtherIdField":"value67","AnotherField":2}]]
```

/status?pretty
```
    "build": {
        "branch": "unknown",
        "build_time": "unknown",
        "commit": "unknown",
        "compiler": "gc",
        "version": "7"
    },
    "cluster": {
        "addr": "192.168.15.46:8905",
        "api_addr": "192.168.15.46:8907",
        "https": "true"
    },
    "http": {
        "auth": "enabled",
        "bind_addr": "192.168.15.46:8907",
        "cluster": {
            "conn_pool_stats": {
                "192.168.15.42:8905": {
                    "idle": 4,
                    "max_open_connections": 64,
                    "open_connections": 4
                },
                "192.168.15.44:8905": {
                    "idle": 4,
                    "max_open_connections": 64,
                    "open_connections": 4
                }
            },
            "local_node_addr": "192.168.15.46:8905",
            "timeout": "30s"
        },
        "queue": {
            "_default": {
                "batch_size": 128,
                "max_size": 1024,
                "sequence_number": 0,
                "timeout": "50ms"
            }
        },
        "tls": {
            "ca_file": "C:\\ProgramData\x\\y\\Database\\DatabaseCertificateAuthority.cer",
            "cert_file": "C:\\ProgramData\x\\y\\Database\\DatabasePublicCertificate.cer",
            "client_auth": "RequireAndVerifyClientCert",
            "enabled": "enabled",
            "key_file": "C:\\ProgramData\\x\\y\\Database\\DatabasePrivateCertificate.pem",
            "next_protos": [
                "h2",
                "http/1.1"
            ]
        }
    },
    "last_backup_time": "2023-05-18T13:56:53.9220773-03:00",
    "node": {
        "current_time": "2023-05-18T13:57:08.0884873-03:00",
        "start_time": "2023-05-18T13:56:34.9209081-03:00",
        "uptime": "33.167576s"
    },
    "os": {
        "executable": "C:\\Program Files\\x\\y\\rqlited.exe",
        "hostname": "server44",
        "page_size": 4096,
        "pid": 4348,
        "ppid": 1204
    },
    "runtime": {
        "GOARCH": "amd64",
        "GOMAXPROCS": 1,
        "GOOS": "windows",
        "num_cpu": 1,
        "num_goroutine": 64,
        "version": "go1.20.2"
    },
    "store": {
        "addr": "192.168.15.46:8905",
        "apply_timeout": "10s",
        "db_applied_index": 275,
        "db_conf": {
            "memory": true,
            "fk_constraints": false
        },
        "dir": "C:\\ProgramData\\x\\y\\Database\\dbname",
        "dir_size": 262144,
        "election_timeout": "10s",
        "fsm_index": 275,
        "heartbeat_timeout": "2s",
        "leader": {
            "addr": "192.168.15.46:8905",
            "node_id": "192.168.15.46"
        },
        "no_freelist_sync": false,
        "node_id": "192.168.15.46",
        "nodes": [
            {
                "id": "192.168.15.42",
                "addr": "192.168.15.42:8905",
                "suffrage": "Voter"
            },
            {
                "id": "192.168.15.44",
                "addr": "192.168.15.44:8905",
                "suffrage": "Voter"
            },
            {
                "id": "192.168.15.46",
                "addr": "192.168.15.46:8905",
                "suffrage": "Voter"
            }
        ],
        "observer": {
            "dropped": 0,
            "observed": 1
        },
        "open": true,
        "raft": {
            "applied_index": 275,
            "bolt": {
                "FreePageN": 0,
                "PendingPageN": 4,
                "FreeAlloc": 16384,
                "FreelistInuse": 48,
                "TxN": 10,
                "OpenTxN": 0,
                "TxStats": {
                    "PageCount": 1160,
                    "PageAlloc": 4751360,
                    "CursorCount": 870,
                    "NodeCount": 836,
                    "NodeDeref": 12,
                    "Rebalance": 0,
                    "RebalanceTime": 0,
                    "Split": 43,
                    "Spill": 872,
                    "SpillTime": 2007900,
                    "Write": 1438,
                    "WriteTime": 1238363700
                }
            },
            "commit_index": 275,
            "fsm_pending": 0,
            "last_contact": 0,
            "last_log_index": 276,
            "last_log_term": 2,
            "last_snapshot_index": 0,
            "last_snapshot_term": 0,
            "latest_configuration": "[{Suffrage:Voter ID:192.168.15.46 Address:192.168.15.46:8905} {Suffrage:Voter ID:192.168.15.44 Address:192.168.15.44:8905} {Suffrage:Voter ID:192.168.15.42 Address:192.168.15.42:8905}]",
            "latest_configuration_index": 0,
            "log_size": 262144,
            "num_peers": 2,
            "protocol_version": 3,
            "protocol_version_max": 3,
            "protocol_version_min": 0,
            "snapshot_version_max": 1,
            "snapshot_version_min": 0,
            "state": "Leader",
            "term": 2,
            "voter": true
        },
        "reap_read_only_timeout": "0s",
        "reap_timeout": "0s",
        "request_marshaler": {
            "compression_batch": 5,
            "compression_size": 150,
            "force_compression": false
        },
        "snapshot_interval": "30s",
        "snapshot_threshold": 8192,
        "sqlite3": {
            "compile_options": [
                "ATOMIC_INTRINSICS=1",
                "COMPILER=gcc-12.2.0",
                "DEFAULT_AUTOVACUUM",
                "DEFAULT_CACHE_SIZE=-2000",
                "DEFAULT_FILE_FORMAT=4",
                "DEFAULT_JOURNAL_SIZE_LIMIT=-1",
                "DEFAULT_MMAP_SIZE=0",
                "DEFAULT_PAGE_SIZE=4096",
                "DEFAULT_PCACHE_INITSZ=20",
                "DEFAULT_RECURSIVE_TRIGGERS",
                "DEFAULT_SECTOR_SIZE=4096",
                "DEFAULT_SYNCHRONOUS=2",
                "DEFAULT_WAL_AUTOCHECKPOINT=1000",
                "DEFAULT_WAL_SYNCHRONOUS=1",
                "DEFAULT_WORKER_THREADS=0",
                "ENABLE_DBSTAT_VTAB",
                "ENABLE_FTS3",
                "ENABLE_FTS3_PARENTHESIS",
                "ENABLE_RTREE",
                "ENABLE_UPDATE_DELETE_LIMIT",
                "MALLOC_SOFT_LIMIT=1024",
                "MAX_ATTACHED=10",
                "MAX_COLUMN=2000",
                "MAX_COMPOUND_SELECT=500",
                "MAX_DEFAULT_PAGE_SIZE=8192",
                "MAX_EXPR_DEPTH=1000",
                "MAX_FUNCTION_ARG=127",
                "MAX_LENGTH=1000000000",
                "MAX_LIKE_PATTERN_LENGTH=50000",
                "MAX_MMAP_SIZE=0x7fff0000",
                "MAX_PAGE_COUNT=1073741823",
                "MAX_PAGE_SIZE=65536",
                "MAX_SQL_LENGTH=1000000000",
                "MAX_TRIGGER_DEPTH=1000",
                "MAX_VARIABLE_NUMBER=32766",
                "MAX_VDBE_OP=250000000",
                "MAX_WORKER_THREADS=8",
                "MUTEX_W32",
                "OMIT_DEPRECATED",
                "OMIT_SHARED_CACHE",
                "SYSTEM_MALLOC",
                "TEMP_STORE=1",
                "THREADSAFE=1"
            ],
            "conn_pool_stats": {
                "ro": {
                    "max_open_connections": 0,
                    "open_connections": 1,
                    "in_use": 0,
                    "idle": 1,
                    "wait_count": 0,
                    "wait_duration": 0,
                    "max_idle_closed": 0,
                    "max_idle_time_closed": 0,
                    "max_lifetime_closed": 0
                },
                "rw": {
                    "max_open_connections": 1,
                    "open_connections": 1,
                    "in_use": 0,
                    "idle": 1,
                    "wait_count": 0,
                    "wait_duration": 0,
                    "max_idle_closed": 0,
                    "max_idle_time_closed": 0,
                    "max_lifetime_closed": 0
                }
            },
            "db_size": 101376,
            "mem_stats": {
                "cache_size": -2000,
                "freelist_count": 1,
                "hard_heap_limit": 0,
                "max_page_count": 1073741823,
                "page_count": 99,
                "page_size": 1024,
                "soft_heap_limit": 0
            },
            "path": ":memory:",
            "ro_dsn": "file:/IolLNtiilhkAgqKgBoLL?mode=ro\u0026vfs=memdb\u0026_txlock=deferred\u0026_fk=false",
            "rw_dsn": "file:/IolLNtiilhkAgqKgBoLL?mode=rw\u0026vfs=memdb\u0026_txlock=immediate\u0026_fk=false",
            "version": "3.39.4"
        },
        "startup_on_disk": false,
        "trailing_logs": 10240
    }
}
```

/debug/vars?pretty
```
"cluster": {
    "num_backup_req": 0,
    "num_execute_req": 28,
    "num_get_node_api_req": 11,
    "num_get_node_api_req_local": 7,
    "num_get_node_api_resp": 11,
    "num_join_req": 0,
    "num_load_req": 0,
    "num_notify_req": 0,
    "num_query_req": 24,
    "num_remove_node_req": 0
},
"cmdline": ["C:\\Program Files\\x\\y\\rqlited.exe", "-node-id", "192.168.15.46", "-http-addr", "192.168.15.46:8907", "-raft-addr", "192.168.15.46:8905", "-join", "https://192.168.15.46:8907,https://192.168.15.42:8907,https://192.168.15.44:8907", "-bootstrap-expect", "3", "-raft-non-voter=false", "-node-ca-cert", "C:\\ProgramData\\x\\y\\Database\\DatabaseCertificateAuthority.cer", "-node-cert", "C:\\ProgramData\\x\\y\\Database\\DatabasePublicCertificate.cer", "-node-verify-client", "-node-key", "C:\\ProgramData\\x\\y\\Database\\DatabasePrivateCertificate.pem", "-http-ca-cert", "C:\\ProgramData\\x\\y\\Database\\DatabaseCertificateAuthority.cer", "-http-cert", "C:\\ProgramData\\x\\y\\Database\\DatabasePublicCertificate.cer", "-http-verify-client", "-http-key", "C:\\ProgramData\\x\\y\\Database\\DatabasePrivateCertificate.pem", "-raft-timeout=2s", "-raft-election-timeout=10s", "C:\\ProgramData\\x\\y\\Database\\dbname"],
"db": {
    "execute_transactions": 0,
    "execution_errors": 0,
    "executions": 273,
    "queries": 129,
    "query_errors": 0,
    "query_transactions": 0
},
"http": {
    "authFail": 0,
    "authOK": 353,
    "backups": 1,
    "executions": 245,
    "joins": 5,
    "leader_not_found": 5,
    "loads": 1,
    "notifies": 5,
    "num_readyz": 2,
    "num_status": 3,
    "queries": 78,
    "queued_executions": 0,
    "queued_executions_failed": 0,
    "queued_executions_leadership_lost": 0,
    "queued_executions_no_leader": 0,
    "queued_executions_not_leader": 0,
    "queued_executions_num_stmts_rx": 0,
    "queued_executions_num_stmts_tx": 0,
    "queued_executions_ok": 0,
    "queued_executions_unknown_error": 0,
    "queued_executions_wait": 0,
    "remote_backups": 0,
    "remote_executions": 0,
    "remote_loads": 0,
    "remote_queries": 0,
    "remote_remove_node": 0
},
"memstats": {
    "Alloc": 5662584,
    "TotalAlloc": 30641440,
    "Sys": 22538088,
    "Lookups": 0,
    "Mallocs": 198782,
    "Frees": 168851,
    "HeapAlloc": 5662584,
    "HeapSys": 11730944,
    "HeapIdle": 4677632,
    "HeapInuse": 7053312,
    "HeapReleased": 2121728,
    "HeapObjects": 29931,
    "StackInuse": 851968,
    "StackSys": 851968,
    "MSpanInuse": 97280,
    "MSpanSys": 114240,
    "MCacheInuse": 1168,
    "MCacheSys": 16352,
    "BuckHashSys": 1461524,
    "GCSys": 7775288,
    "OtherSys": 587772,
    "NextGC": 8465768,
    "LastGC": 1684429027579149000,
    "PauseTotalNs": 502600,
    "PauseNs": [0, 0, 0, 0, 0, 0, 502600, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    "PauseEnd": [1684428998656269400, 1684428999531524800, 1684429009859575000, 1684429013430457800, 1684429017393421600, 1684429025646918100, 1684429026133200100, 1684429026882950200, 1684429027579149000, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    "NumGC": 9,
    "NumForcedGC": 0,
    "GCCPUFraction": 0.0003758392890398497,
    "EnableGC": true,
    "DebugGC": false,
    "BySize": [{
            "Size": 0,
            "Mallocs": 0,
            "Frees": 0
        }, {
            "Size": 8,
            "Mallocs": 6070,
            "Frees": 3998
        }, {
            "Size": 16,
            "Mallocs": 44811,
            "Frees": 37389
        }, {
            "Size": 24,
            "Mallocs": 22028,
            "Frees": 18643
        }, {
            "Size": 32,
            "Mallocs": 13947,
            "Frees": 11669
        }, {
            "Size": 48,
            "Mallocs": 20973,
            "Frees": 16788
        }, {
            "Size": 64,
            "Mallocs": 12250,
            "Frees": 10719
        }, {
            "Size": 80,
            "Mallocs": 15280,
            "Frees": 12870
        }, {
            "Size": 96,
            "Mallocs": 10752,
            "Frees": 9106
        }, {
            "Size": 112,
            "Mallocs": 3927,
            "Frees": 3534
        }, {
            "Size": 128,
            "Mallocs": 4100,
            "Frees": 3614
        }, {
            "Size": 144,
            "Mallocs": 3272,
            "Frees": 2773
        }, {
            "Size": 160,
            "Mallocs": 356,
            "Frees": 220
        }, {
            "Size": 176,
            "Mallocs": 884,
            "Frees": 345
        }, {
            "Size": 192,
            "Mallocs": 1111,
            "Frees": 947
        }, {
            "Size": 208,
            "Mallocs": 600,
            "Frees": 415
        }, {
            "Size": 224,
            "Mallocs": 775,
            "Frees": 622
        }, {
            "Size": 240,
            "Mallocs": 398,
            "Frees": 239
        }, {
            "Size": 256,
            "Mallocs": 995,
            "Frees": 762
        }, {
            "Size": 288,
            "Mallocs": 1596,
            "Frees": 1446
        }, {
            "Size": 320,
            "Mallocs": 832,
            "Frees": 616
        }, {
            "Size": 352,
            "Mallocs": 2094,
            "Frees": 1754
        }, {
            "Size": 384,
            "Mallocs": 146,
            "Frees": 77
        }, {
            "Size": 416,
            "Mallocs": 143,
            "Frees": 32
        }, {
            "Size": 448,
            "Mallocs": 297,
            "Frees": 225
        }, {
            "Size": 480,
            "Mallocs": 21,
            "Frees": 10
        }, {
            "Size": 512,
            "Mallocs": 2449,
            "Frees": 2199
        }, {
            "Size": 576,
            "Mallocs": 9239,
            "Frees": 9157
        }, {
            "Size": 640,
            "Mallocs": 168,
            "Frees": 94
        }, {
            "Size": 704,
            "Mallocs": 92,
            "Frees": 65
        }, {
            "Size": 768,
            "Mallocs": 55,
            "Frees": 41
        }, {
            "Size": 896,
            "Mallocs": 172,
            "Frees": 93
        }, {
            "Size": 1024,
            "Mallocs": 125,
            "Frees": 92
        }, {
            "Size": 1152,
            "Mallocs": 102,
            "Frees": 77
        }, {
            "Size": 1280,
            "Mallocs": 76,
            "Frees": 54
        }, {
            "Size": 1408,
            "Mallocs": 310,
            "Frees": 197
        }, {
            "Size": 1536,
            "Mallocs": 68,
            "Frees": 58
        }, {
            "Size": 1792,
            "Mallocs": 101,
            "Frees": 81
        }, {
            "Size": 2048,
            "Mallocs": 122,
            "Frees": 82
        }, {
            "Size": 2304,
            "Mallocs": 89,
            "Frees": 47
        }, {
            "Size": 2688,
            "Mallocs": 95,
            "Frees": 48
        }, {
            "Size": 3072,
            "Mallocs": 17,
            "Frees": 8
        }, {
            "Size": 3200,
            "Mallocs": 3,
            "Frees": 1
        }, {
            "Size": 3456,
            "Mallocs": 12,
            "Frees": 8
        }, {
            "Size": 4096,
            "Mallocs": 602,
            "Frees": 472
        }, {
            "Size": 4864,
            "Mallocs": 34,
            "Frees": 24
        }, {
            "Size": 5376,
            "Mallocs": 23,
            "Frees": 9
        }, {
            "Size": 6144,
            "Mallocs": 7,
            "Frees": 3
        }, {
            "Size": 6528,
            "Mallocs": 3,
            "Frees": 3
        }, {
            "Size": 6784,
            "Mallocs": 0,
            "Frees": 0
        }, {
            "Size": 6912,
            "Mallocs": 4,
            "Frees": 4
        }, {
            "Size": 8192,
            "Mallocs": 30,
            "Frees": 18
        }, {
            "Size": 9472,
            "Mallocs": 10,
            "Frees": 6
        }, {
            "Size": 9728,
            "Mallocs": 0,
            "Frees": 0
        }, {
            "Size": 10240,
            "Mallocs": 1,
            "Frees": 0
        }, {
            "Size": 10880,
            "Mallocs": 0,
            "Frees": 0
        }, {
            "Size": 12288,
            "Mallocs": 5,
            "Frees": 4
        }, {
            "Size": 13568,
            "Mallocs": 0,
            "Frees": 0
        }, {
            "Size": 14336,
            "Mallocs": 1,
            "Frees": 0
        }, {
            "Size": 16384,
            "Mallocs": 8,
            "Frees": 5
        }, {
            "Size": 18432,
            "Mallocs": 1,
            "Frees": 1
        }
    ]
},
"mux": {
    "num_connections_handled": 8,
    "num_unregistered_handlers": 0
},
"proto": {
    "num_compressed_bytes": 1850,
    "num_compressed_requests": 6,
    "num_compression_misses": 0,
    "num_precompressed_bytes": 59149,
    "num_requests": 273,
    "num_uncompressed_bytes": 56710,
    "num_uncompressed_requests": 267
},
"queue": {
    "num_flush": 0,
    "num_timeout": 0,
    "statements_rx": 0,
    "statements_tx": 0
},
"store": {
    "failed_heartbeat_observed": 0,
    "leader_changes_dropped": 0,
    "leader_changes_observed": 0,
    "nodes_reaped_failed": 0,
    "nodes_reaped_ok": 0,
    "num_backups": 1,
    "num_compressed_commands": 6,
    "num_ignored_joins": 0,
    "num_joins": 0,
    "num_loads": 1,
    "num_recoveries": 0,
    "num_removed_before_joins": 0,
    "num_restores": 0,
    "num_snapshots": 0,
    "num_uncompressed_commands": 267,
    "snapshot_create_duration": 0,
    "snapshot_db_ondisk_size": 0,
    "snapshot_db_serialized_size": 0,
    "snapshot_persist_duration": 0
}
}
```

/nodes?pretty
```
    "192.168.15.42": {
        "api_addr": "https://192.168.15.42:8907",
        "addr": "192.168.15.42:8905",
        "reachable": true,
        "leader": false,
        "time": 0.0013598
    },
    "192.168.15.44": {
        "api_addr": "https://192.168.15.44:8907",
        "addr": "192.168.15.44:8905",
        "reachable": true,
        "leader": false
    },
    "192.168.15.46": {
        "api_addr": "https://192.168.15.46:8907",
        "addr": "192.168.15.46:8905",
        "reachable": true,
        "leader": true
    }
}
```

**Comment by otoolep** [comment]
Thanks for your report -- obviously any report of data being written, but not being persisted, is serious. Interestingly I have seen some strange behaviour on Windows that I have not been able to track down (you referenced it) but only with Queued Writes, which you are not using.

Thanks for the information you supplied, but I'm missing the most important information -- how are you actually writing the information? Are you using a client library? Show me the API calls you make. Are you using`curl` calls? Without precise details on how you are using the API, it's difficult to know if you're doing something wrong (like not using the API correctly, you're ignore an error code, or not building your request properly). Are you sending all your requests to the same node? Are they going directly to the Leader (it shouldn't matter, just wondering if you know -- the metrics you supplied indicate you are).

Using `/debug/vars` will be key here. It will tell you exactly how many requests rqlite has received, how many SQL statements it wrote to the database, and so on. You need to find a discrepancy between what your code is telling you you wrote, and what rqlite is telling you it received.

- https://github.com/rqlite/rqlite/blob/master/http/service.go#L417 -- tells you how many times the /db/execute endpoint was hit. This number is 245 in the information you provided. Does this line up with your client code?
- https://github.com/rqlite/rqlite/blob/master/db/db.go#L419 - tells you how many SQL statements were actually written to the database. This number is 273 in the information you provided, again how many SQL statements do you think you inserted?

rqlite is heavily instrumented, and everything we need to figure out if writes are being dropped is being measured by rqlite. What you need to find is a clear disprepancy between the counts your client code is keeping, and the counts rqlite is showing you on /debug/vars.

**Comment by filipesbragio** [comment]
Thanks for the quick response!

We're using HttpClient from C# to connect to the rqlite endpoint. What you're seeing in those logs is a part of the initial setup of the application, part of which is adding default data to the database. We are connecting to the leader although if leadership changes partway through this initial setup we would continue sending it to the previous leader until initialization finishes.

I added logs to check how many executions we made and started the system with a clean database until the issue happened again. This time we're also missing 4 entries in the database although they're not the same 4 entries. My logs do match the /debug/vars logs with 245 times client requests and rqlite receiving 245 hits on /db/execute. 

I'm not sure I understand what the 273 number represents(271 during this last try), does it mean the 245 executions we sent + 28 executions rqlite might have made internally during startup or when we're loading a pre-existing database? I'm unsure where I might search for those 28 executions that are unaccounted for. 

We're sending a POST request to "/db/execute?timings" like so:

```
var request = new HttpRequestMessage(httpMethod, requestUri);
if (!string.IsNullOrWhiteSpace(content))
{
	request.Content = new StringContent(content, Encoding.UTF8, "application/json");
}
using (var response = await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead).ConfigureAwait(false))
{
	log.Error(content);
	log.Error(response.IsSuccessStatusCode);
	var responseContent = await response.Content.ReadAsStringAsync();
	log.Error(responseContent);
	if (response.IsSuccessStatusCode)
	{
		return JsonConvert.DeserializeObject<T>(responseContent);
	}
	else if (retryAttempt++ < maxRetries)
	{
		Task.Delay(retryDelay).Wait();
		return await client.DoSendTyped<T>(httpMethod, requestUri, content, retryAttempt, retryDelay);
	}
	log.Error($"Error when sending request, [{response.StatusCode}] - [{responseContent}]");
	return default;
}
```

**Comment by otoolep** [comment]
>We are connecting to the leader although if leadership changes partway through this initial setup we would continue sending it to the previous leader until initialization finishes.

Are you sure you're handling any Leadership changes correctly? A write cannot succeed if there is no leader on the cluster. Your `/debug/vars` show non-zero counts for `leader_not_found`, which might indicate that you hit that condition. That said, a leader going down while writing should result in errors getting send back to the client. Are you looking for that correctly? It's important to check both the HTTP status code and  the *body* of the response for the `error` key being set - just checking the HTTP return code is not enough. You can get HTTP 200, but the SQL statement still fails to be written. I don't see any checking for an "error" key in your code above.

https://rqlite.io/docs/api/api/#handling-errors

I will say that the problem I have seen occasionally during QueuedWrites testing on Windows does seem to involve writing during Leadership changes. Do you see a Leader election happening every time you hit this issue? It should be handled correctly, but maybe there is something up with Windows because I've not seen this issue (nor hand any reports) of issues with other OSes.

**Comment by otoolep** [comment]
As for the extra executes, they could be loads, create table statements at the start. Not sure without knowing more.

**Comment by filipesbragio** [comment]
We deserialize the body into a class that has the results and inside each there's an error field, we don't check it during that code snippet we just sent but we do check for it elsewhere. That shouldn't be the case though since in that code snippet we're logging the body and there don't seem to be any errors in the logs.

During this latest attempt we had 3 instances of leader_not_found which I can find in my logs during startup while rqlite was still initializing, it doesn't seem like there were any leadership changes happening partway through the requests where the issue happens.

I did try increasing the number of processors in my VM's from 1 to 2 to see if that might help but I can still reproduce it somewhat often. That was my main suspicion as to why it might be related to 1123 and why it wasn't reproducible through our tests but it doesn't seem to make sense if it still happens after scaling the VM's

**Comment by filipesbragio** [comment]
I believe I've found the issue on our code, there was code running in parallel which shouldn't and they both were configuring that same table. I'm going to make sure that's what's happening and close this as soon as I'm certain(likely tomorrow morning). 

Thanks for your help/extra information and I'm so sorry to have wasted your time.

**Comment by otoolep** [comment]
Great, let me know how it goes.

And it wasn't a waste of my time -- I'm actually improving some of the metrics as result of this bug, and reporting a potentially serious issue with rqlite is important. I appreciated the report.
